File: tutorials/java/01-hello-world/src/test/java/hello/HelloTest.java
package hello;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class HelloTest {

    @Test
    void greetWithName() {
        assertEquals("Hello, Alice!", Hello.greet("Alice"));
    }

    @Test
    void greetWithWorld() {
        assertEquals("Hello, World!", Hello.greet("World"));
    }
}

File: tutorials/elixir/01-hello-world/test/hello_test.exs
defmodule HelloTest do
  use ExUnit.Case, async: true

  test "greet/0 returns a generic greeting" do
    assert Hello.greet() == "Hello, world!"
  end

  test "greet/1 returns a personalized greeting" do
    assert Hello.greet("Alice") == "Hello, Alice!"
  end

  test "greet/1 handles an empty string" do
    assert Hello.greet("") == "Hello, !"
  end
end
